### DISCUSSION TITLE
spark-submit  error：java.io.FileNotFoundException: libroken.so.18

### DISCUSSION THREAD

**Discussion by mgtu-code** [apache/gluten#2512] [Q&A]
1、spark-submit --conf spark.plugins=io.glutenproject.GlutenPlugin --conf spark.gluten.loadLibFromJar=true --conf spark.shuffle.manager=org.apache.spark.shuffle.sort.ColumnarShuffleManager --jars "${RootDir}temp/gluten-velox-bundle-spark3.2_2.12-ubuntu_20.04-1.0.0.jar"
2、report error：java.io.FileNotFoundException: libroken.so.18

env：
spark 3.2.2
unbutu 20.04 

how can i solve it

**Comment by philo-he** [comment]
Hi @mgtu-code, please set false for `spark.gluten.loadLibFromJar`.The loading native lib behavior is not applicable for the posted latest release jar (native lib already statically linked).

BTW, could you let us know your company name?

**Reply by mgtu-code** [reply]
thx，set spark.gluten.loadLibFromJar=false  solve it，But another error occurs, as follows:

ERROR vectorized.JniLibLoader: Unload native library error: 
java.lang.ClassCastException: class java.util.concurrent.ConcurrentHashMap cannot be cast to class java.util.Vector (java.util.concurrent.ConcurrentHashMap and java.util.Vector are in module java.base of loader 'bootstrap')

I am an engineer of mobvista

**Comment by philo-he** [comment]
Could you file a github issue with more detailed log? Thanks!
